### ISSUE TITLE
Bug: "UseStaticRegexIsMatchAnalyzer" causes an exception (CC0081)

### ISSUE THREAD

**Issue by ElStefan** [code-cracker/code-cracker#822]
Hi,

while working with regular expressions in C#, an exception occured in the error list caused by code-cracker.

After some playing around with the code, i found the suspicous part in my code which is causing the exception.

````csharp
public class RegexTestClass
{
    private TestModel testModel;

    private void Test(string text)
    {
        if (testModel.Regex.IsMatch(text))
        {
            return;
        }
    }
}

public class TestModel
{
    public System.Text.RegularExpressions.Regex Regex { get; set; }
}
````

The exception (only got it in german):

````
Analyzer 'CodeCracker.CSharp.Performance.UseStaticRegexIsMatchAnalyzer' threw 
an exception of type 'System.InvalidCastException' with message
'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.IdentifierNameSyntax'.'.	
````

A more complex scenario:

````csharp
public class RegexTestClass
{
    private C c;

    private void Test(string text)
    {
        if (c.TestModel.Regex.IsMatch(text))
        {
            return;
        }
    }
}
public class C
{
    public TestModel TestModel { get; set; }
}
public class TestModel
{
    public System.Text.RegularExpressions.Regex Regex { get; set; }
}
````

````
Active	Analyzer 'CodeCracker.CSharp.Performance.UseStaticRegexIsMatchAnalyzer' threw the following exception:
'Exception occurred with following context:
Compilation: ConsoleApplication1
SyntaxTree: c:\temp\ConsoleApplication1\ConsoleApplication1\Program.cs
SyntaxNode: testModel.Regex.IsMatch(text) [InvocationExpressionSyntax]@[439..468) (21,16)-(21,45)

System.InvalidCastException: Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.IdentifierNameSyntax'.
   at CodeCracker.CSharp.Performance.UseStaticRegexIsMatchAnalyzer.Analyzer(SyntaxNodeAnalysisContext context) in C:\p\code-cracker\src\CSharp\CodeCracker\Performance\UseStaticRegexIsMatchAnalyzer.cs:line 44
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass42_1`1.<ExecuteSyntaxNodeAction>b__1()
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info)
````

I guess the problematic line is in [here](https://github.com/code-cracker/code-cracker/blob/master/src/CSharp/CodeCracker/Performance/UseStaticRegexIsMatchAnalyzer.cs) on line 44

Thanks, and thanks again for fixing the other issue I submitted!

The PR has to go on the `v1.0.x` branch.

**Comment by carloscds** [comment]
@ElStefan Can you check if you are running last version of CodeCracker ? I can't reproduce this error.

**Comment by ElStefan** [comment]
I was using the master branch to figure it out, so i think it should be the latest version.
I have to admit, when I tried to reproduce it again today, it didn't work the second time.
So I wrote a unit test according to your existing tests and it's definitely throwing an exception every time now.


```
[Fact]
        public async Task ExceptionOnUsingThisKeyword()
        {
            const string testException = @"
    using System.Text.RegularExpressions;

    namespace ConsoleApplication1
    {
        public class RegexTestClass
        {
            private Regex expression;

            private void Test(string text)
            {
                if (this.expression.IsMatch(text))
                {
                    return;
                }
            }
        }
    }";
            await VerifyCSharpHasNoDiagnosticsAsync(testException);
        }

**Comment by ElStefan** [comment]
Okay i got it now. It's happening if "IsMatch" is called in a nested expression.

````csharp
public class RegexTestClass
{
	private TestModel testModel;

	private void Test(string text)
	{
		if (testModel.Regex.IsMatch(text))
		{
			return;
		}
	}
}

public class TestModel
{
	public Regex Regex { get; set; }
}
````
( ..and now i learned how to markdown code in github correctly :) )

**Comment by giggio** [comment]
Bug confirmed, here is the full stack trace:

````
Analyzer 'CodeCracker.CSharp.Performance.UseStaticRegexIsMatchAnalyzer' threw 
an exception of type 'System.InvalidCastException' with message
'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.IdentifierNameSyntax'.'.	
````

````
Active	Analyzer 'CodeCracker.CSharp.Performance.UseStaticRegexIsMatchAnalyzer' threw the following exception:
'Exception occurred with following context:
Compilation: ConsoleApplication1
SyntaxTree: c:\temp\ConsoleApplication1\ConsoleApplication1\Program.cs
SyntaxNode: testModel.Regex.IsMatch(text) [InvocationExpressionSyntax]@[439..468) (21,16)-(21,45)

System.InvalidCastException: Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.IdentifierNameSyntax'.
   at CodeCracker.CSharp.Performance.UseStaticRegexIsMatchAnalyzer.Analyzer(SyntaxNodeAnalysisContext context) in C:\p\code-cracker\src\CSharp\CodeCracker\Performance\UseStaticRegexIsMatchAnalyzer.cs:line 44
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass42_1`1.<ExecuteSyntaxNodeAction>b__1()
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info)
````

**Comment by giggio** [comment]
Closed on #833
